r1cs/gadgets/std/src/fields/nonnative/tests.rs

```diff
@@ -1602,32 +2039,62 @@ macro_rules! nonnative_test {
  #[cfg(feature = "mnt4_753")]
  use algebra::fields::mnt4753::{Fq as Mnt4753Fq, Fr as Mnt4753Fr};
 
+ #[cfg(feature = "bls12_377")]
+ use algebra::fields::bls12_377::{Fq as Bls12377Fq, Fr as Bls12377Fr};
+
+// tests over tweedle Fr
+
 #[cfg(feature = "tweedle")]
 nonnative_test!(TweedleFq_over_Fr, TweedleFq, TweedleFr);
 
+#[cfg(all(feature = "tweedle", feature = "ed25519"))]
+nonnative_test!(ed25519Fq_over_TweedleFr, ed25519Fq, TweedleFr);
+
+#[cfg(all(feature = "tweedle", feature = "secp256k1"))]
+nonnative_test!(secp256k1_over_TweedleFr, secp256k1Fq, TweedleFr);
+
+#[cfg(all(feature = "tweedle", feature = "bn_382"))]
+nonnative_test!(Bn382Fr_over_TweedleFr, Bn382Fr, TweedleFr);
+
+#[cfg(all(feature = "tweedle", feature = "mnt4_753"))]
```

Comment:
Are these mnt4 tests necessary ? They are quite heavy and we will remove the curve anyway

Reply:
The more fields we test the better.